dashboard/test/models/user_test.rb

```diff
@@ -1897,6 +1897,97 @@ def update_primary_contact_info_fails_safely_for(user, *params)
     assert_equal original_primary_contact_info, user.primary_contact_info
   end
 
+  def upgrade_to_personal_login_params(args = {})
```

Comment:
I'm not sure if this is more Ruby-idiomatic, but double-splat and `merge` would let callers override any defaults in the hash.
```
2.5.0 :001 > def many_args(**args)
2.5.0 :002?>   {
2.5.0 :003 >       a: 'default_a',
2.5.0 :004 >       b: 'default_b',
2.5.0 :005 >       }.merge(args)
2.5.0 :006?>   end
 => :many_args 
2.5.0 :007 > many_args(b: 'new_b', c: 'new_c')
 => {:a=>"default_a", :b=>"new_b", :c=>"new_c"} 
```